### PR TITLE
dl: optimize cpu matmul

### PR DESCRIPTION
**Pull request by piscisaureus**
On my machine, on the order of 100x speedup.
To try: `PROPEL=dl ts-node matmul_bench.ts`.
I'm not sure where the benchmark should go.

### CODE REVIEW

File: src/dl/math/backends/backend_cpu.ts [hunk 1]
```diff
@@ -337,30 +337,37 @@ export class MathBackendCPU implements MathBackend {
...
-        transposedGetter;
-    const bGetter = (bOrientation === MatrixOrientation.REGULAR) ?
-        normalGetter :
-        transposedGetter;
-    const values = new Float32Array(leftDim * rightDim);
-    let index = 0;
-    for (let i = 0; i < leftDim; ++i) {
-      for (let j = 0; j < rightDim; ++j) {
+    const aValues = a.dataSync();
+    const bValues = b.dataSync();
```
Comment (ry) [inline src/dl/math/backends/backend_cpu.ts]: I’m concerned about the sync here. Does this patch speed up testNNExample?

Comment (piscisaureus) [reply src/dl/math/backends/backend_cpu.ts]: Yes, a lot.
Also this code is cpu only - dataSync is basically a no-op.

Comment (ry) [reply src/dl/math/backends/backend_cpu.ts]: oh right. can confirm 100x speed up on mac.

File: matmul_bench.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+import { tensor } from "./src/api";
+
+const t1 = tensor(new Array(100).fill(new Array(100).fill(1)));
+const t2 = tensor(new Array(100).fill(new Array(100).fill(2)));
```
Comment (ry) [inline matmul_bench.ts]:
```
const t1 = ones([100, 100]);
const t2 = ones([100, 100]).add(1);
```

Comment (piscisaureus) [reply matmul_bench.ts]: fixed

**Pull request merged**